vivarium/framework/population.py

```diff
@@ -10,18 +10,31 @@
 from .event import emits, Event
 
 uses_columns = resource_injector('population_system_population_view')
-uses_columns.__doc__ = """Mark a function as a user of columns from the population table. If the
-function is also an event listener then the Event object which it receives will be transformed into a
-PopulationEvent. Otherwise the function will have a configured PopulationView injected into its arguments.
+uses_columns.__doc__ = \
+"""Mark a function as a user of columns from the population table. If
+the function is also an event listener then the Event object which it
+receives will be transformed into a PopulationEvent. Otherwise the
+function will have a configured PopulationView injected into its
+arguments.
 
 Parameters
 ----------
 columns : [str]
-          A list of column names which the function will need to read or write.
+          A list of column names which the function will need to read
+          or write.
+
 query   : str
-          A filter in pandas query syntax which should be applied to the population before it made accessible
-          to this function. This effects both the ``population`` and the ``index`` attributes of PopulationEvents
+          A filter in pandas query syntax which should be applied to
+          the population before it made accessible to this
+          function. This effects both the ``population`` and the
+          ``index`` attributes of PopulationEvents
 """
+from inspect import signature, Parameter, Signature
+uses_columns.__signature__ = \
```

Comment:
If you want a more tool friendly signature I wouldn't do this. Rather I would wrap the generated function like this:
`
_uses_columns = resource_injector('population_system_population_view')
def uses_columns(columns, query):
    '''DOCSTRING'''
   return _uses_columns(columns, query)
`

I haven't actually tried that but it should work and give you the behavior you want.